src/lib.rs

```diff
@@ -1927,8 +1941,16 @@ impl Connection {
 
         // TODO: implement backpressure based on peer's flow control
 
+        let was_writable = stream.writable();
+
         stream.send.push_slice(buf, fin)?;
 
+        // Push stream to the writable queue, but only if it wasn't already
+        // queued..
+        if !was_writable {
```

Comment:
No need to check `stream.writable()`?

Reply:
The original thinking was that since we are adding new data the stream would be always writable, but now that I think more about it, the stream might not have flow control credits so we should indeed check for `stream.writable()` here as well.

Reply:
Fixed, thanks.